api/registrations/serializers.py

```diff
@@ -343,11 +343,14 @@ def update(self, registration, validated_data):
                     raise exceptions.PermissionDenied()
             else:
                 raise exceptions.ValidationError('Registrations can only be turned from private to public.')
+        if not user_is_admin and (validated_data.get('retraction') or validated_data.get('is_retracted')):
+            raise exceptions.PermissionDenied()
         if validated_data.get('retraction') and not validated_data.get('is_retracted'):
             raise exceptions.ValidationError(
                 'You cannot provide a withdrawal_justification without a concurrent withdrawal request.',
             )
-        if is_truthy(validated_data.get('is_retracted', False)):
+        is_retracted = validated_data.get('is_retracted', None)
```

Comment:
I would just suggest a little bit of cleanup on this method - you define `is_retracted` here, but your above code would have benefitted from this variable.  I also might rearrange this to make it a little more readable, just personal opinion, welcome other suggestions
```python
if either retraction field was in payload:
   if user is an admin:
           - validation checks to make sure we can proceed with retraction
          -  retract registration
   elif:
         raise 403 because they don't have permission
```